### PR TITLE
RFC: Clarify when we promise to be on-call

### PR DESCRIPTION
**Pull request by sweir27**
Originally paired with @orta on this, who gets all the credit for the fancy visualization. 😄 

## Proposal
A "Service-Level Objective" for the hours during which engineers are expected to be monitoring for incidents during their on-call schedule.

_See doc below for the details._

## Reasoning
There is ambiguity right now around *when* engineers are expected to respond in a timely manner to incidents. The docs currently say:

> While we would like to have 24/7 on-call coverage, we recognize that we don't yet have a good solution for notifying engineers during non-work hours. Do your best to keep an eye out for issues, and lean on those in more favorable timezones where appropriate.

Given we do not have a way to accurately notify engineers of issues aside from slack (which does not have great support for "waking people up"), we cannot claim to have 24/7 support.

This proposal aims to capture *what we already do* in order to reduce ambiguity, and therefore stress, around expectations.

When we have better support for notifications (i.e. through a homegrown service or by integrating with popular services like PagerDuty or OpsGenie), we will be able to revisit this SLO.

## Exceptions
Since we're communicating an SLO in EST, we'll need to make sure that there is at least one person in that timezone on-call at any given time. For now, we can hold each other accountable in lieu of an automated solution to this.

## Additional Context
Service Level Objective Context/Definition: https://landing.google.com/sre/sre-book/chapters/service-level-objectives/

### CODE REVIEW

**Comment by orta** [comment]
We don't explicitly say EST in the SLO doc - probably worth adding that somewhere I think

**Comment by mzikherman** [comment]
I'm 👍 this and anything else to reduce ambiguity and ensure a more consistent on-call process.

I was expecting this RFC to clarify a few things and set out some clear guidelines around the ambiguous parts, but then came across:

> This proposal aims to capture what we already do in order to reduce ambiguity, and therefore stress, around expectations.

I sort of feel like there _is_ some ambiguity still left in our current process, so just capturing what we do (which I think we've done already), might not be sufficient without _actually_ clarifying things.

Here's a specific question I would love to see answered as part of an effort to clarify process.

What is the **exact** time(s) during which on-call people are expected, and not expected to be available? For instance, what if I'm on call and I started work @ 9am that morning, and so by 5pm I'm done. Am I just good to go until 9am the next morning (since indeed, we are now outside of my normal working hours)? What if I have plans that night- do I need to notify my on-call partner and try to have one of us around on our computer? What if my plans involve sitting on my couch- am I now obligated to have my phone w/ me or to be online? Let's say an incident occurs @ 7:30pm, most people are gone for the day but it's certainly not the middle of the night just yet. Who's actually responsible for this? How are people being notified outside of working hours when they're not at their computer?

We've [rightfully] receded from a promise of 24 hour support, but there's still ambiguity left.

Here's a link to the playbook: https://github.com/artsy/README/tree/42868f2c356fcd36f21325d43350a092c41ac6de/playbooks/support , and a relevant quote:

> During work hours, on-call engineers are responsible for responding to issues in the #incidents slack channel 🔒 in a timely fashion. Outside of work hours, engineers are only responsible for downtime issues.

A few lines later:

>...we recognize that we don't yet have a good solution for notifying engineers during non-work hours. Do your best to keep an eye out for issues...

To me, this is a stressful combination of statements. One the one hand, we're responsible for downtime issues outside of work hours, but on the other hand the only guidance is to "do your best to keep an eye out for issues". I personally would love some clarity around what exactly 'work hours' constitute, and how downtime issues outside of those hours can be brought to people's attention. I don't think it's fair to ask people to be responsible for something outside of work hours without actually having a proper system of escalating a downtime incident to a phone-call or page. I would personally vote for removing that ambiguity by **rolling back incident coverage to just EST business hours 9am-5pm, with people in other timezones who are on-call/available a bonus**. A requirement for on-call beyond that (such as a night shift, weekends, or anything outside of those hours), should come with not only adequate compensation but also an industry standard system for proper alerting.

But...all that being said, I am very much 👍 any ticket attempting to clarify on-call ambiguity! The above is just my opinion. Definitely curious to hear what other people think about remaining ambiguity in the process.

**Comment by jonallured** [comment]
Wow, very well said @mzikherman!! In order to feel less stress while on-call, I've basically been operating as you describe: during 9-5, I'm watching #incidents and very on-call. Outside those hours, I've probably got my phone handy, but I'm not very on-call. Reflecting on my recent shift, I did check in with my partners a bit when I wouldn't be around, but if downtime had occurred outside of 9-5, I'm not sure I would have noticed right away. Just trying to be honest here!

**Comment by SamRozen** [comment]
Thanks for raising this. It's been a pain point for a while, it's great to be more explicit about this.
One additional question, what are commitments for live auctions?

File: playbooks/support/on-call-slo.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# On-Call SLO
+
+This document describes a "Service-Level Objective" for the hours during with engineers are expected to be monitoring for incidents during their on-call schedule.
+
+![visualization of our on-call SLO](images/on-call-slo.jpg)
+
+## What we promise
```
Comment (dleve123) [inline playbooks/support/on-call-slo.md]: Can we clarify ET?

File: playbooks/support/on-call-slo.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# On-Call SLO
+
+This document describes a "Service-Level Objective" for the hours during with engineers are expected to be monitoring for incidents during their on-call schedule.
+
+![visualization of our on-call SLO](images/on-call-slo.jpg)
```
Comment (dleve123) [inline playbooks/support/on-call-slo.md]: link is broken in the rendered Github diff

**Comment by mzikherman** [comment]
Omg lol I thought this was an RFC issue and not a PR 😅 , I didn't actually see the diff, or the diagram of weekdays/weekends until @dleve123 started commenting. And in fact I do see it's more explicit about the scheduling which is one of the main things I was asking for 👍

File: playbooks/support/on-call-slo.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# On-Call SLO
+
+This document describes a "Service-Level Objective" for the hours during with engineers are expected to be monitoring for incidents during their on-call schedule.
```
Comment (dleve123) [inline playbooks/support/on-call-slo.md]: I believe the proper grammar here is "during which"?

File: playbooks/support/on-call-slo.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+![visualization of our on-call SLO](images/on-call-slo.jpg)
+
+## What we promise
+- Work days (excludes holidays)
```
Comment (dleve123) [inline playbooks/support/on-call-slo.md]: Thoughts on linking to the [official People Ops policy doc](https://docs.google.com/document/d/16FQ0nKIJ_Eb11kWQd70neB7V3i51ixUsf-CB5f_rAMo/edit#heading=h.d93f947ory) for ease of reference when determining if a certain day is a holiday?

File: playbooks/support/on-call-slo.md (hunk 3 above)
Comment (dleve123) [reply playbooks/support/on-call-slo.md]: also, thoughts on linking to Google's resource on SLOs (https://landing.google.com/sre/sre-book/chapters/service-level-objectives/) considering this term isn't widely used at Artsy yet?

**Comment by ansor4** [comment]
Agree with @SamRozen that live auctions probably deserve special attention, but not sure how much support we want to promise to live auctions occurring during limited or partial support times. If it's the same amount of support, we probably want to explicitly say so.

File: playbooks/support/on-call-slo.md (hunk 3 above)
Comment (dblandin) [reply playbooks/support/on-call-slo.md]: Love these clarifications so far! Thanks so much for writing these up! :heart: 

I might rephrase this line to focus on the response time. Here we're setting expectations around when on-call engineers should be available to respond to incidents, but the objectives themselves might be better set as explicit response time thresholds.

For example:
- Incidents should have a response time of under 10m, during hours of full support
- Incidents should have a response time of under 30m, during hours of partial support

We might also define what it means to initially respond to an incident and how that time is measured as part of the SLO. To represent our current process, we might say that to respond the incident, you acknowledge it in the #incidents slack channel and create a Jira Ops incident ticket, transitioned to an active state.

These SLOs  would target the time between first reported in #incidents and a Jira Ops ticket created and transitioned to an active status by an engineer.

**Comment by sweir27** [comment]
Thank you all for the comments! Will update based on @dleve123 's comments above.

Re: live auctions, I can add a note here but this won't attempt to change how we currently deal with them. Downtime issues which affect the rest of Artsy will likely affect live auctions, and will be dealt with within the times outlined here. If there's a special need to monitor a high-profile auction, that should be scheduled ahead of time with a product team and not fall to the on-call team members.

As we start supporting more and more transactions from our global buyers, we'll likely want to expand our coverage anyways (but as stated in the description, this is just an attempt to clarify what we currently offer, not make assertions about where we may be someday).

File: playbooks/support/on-call-slo.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# On-Call SLO
+
+This document describes a
```
Comment (sweir27) [inline playbooks/support/on-call-slo.md]: @dblandin updated this doc to be more explicit about our SLO (as referring to our response time). Is it clearer?

Comment (dblandin) [reply playbooks/support/on-call-slo.md]: Yes, that looks great!! :tada:

**Pull request merged**